### ISSUE TITLE
error: `tpp.identity` op fails to verify broadcasting rules

### ISSUE THREAD

**Issue by rengolin** [plaidml/tpp-mlir#353]
As exposed in #342, TPP identity is being mapped wrongly in existing models (as well as the in-development MLP generator, following the same patterns).

Here's the error:
```
test/Models/multi-head-attention.mlir:125:12: error: 'tpp.identity' op fails to verify broadcasting rules
    %102 = linalg.generic {indexing_maps = [affine_map<(d0, d1, d2, d3) -> (d0, d3, d1, d2)>, affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%98 : tensor<32x8x2x64xf32>) outs(%101 : tensor<32x2x64x8xf32>) {
           ^
test/Models/multi-head-attention.mlir:125:12: note: see current operation: "tpp.identity"(%62, %63) : (memref<8x64xf32, affine_map<(d0, d1)[s0] -> (d0 * 128 + d1 + s0)>>, memref<64x8xf32, affine_map<(d0, d1)[s0] -> (d0 * 8 + d1 + s0)>>) -> ()
```

I believe this is the identity:
```
    %101 = tensor.empty() : tensor<32x2x64x8xf32>
    %102 = linalg.generic {indexing_maps = [affine_map<(d0, d1, d2, d3) -> (d0, d3, d1, d2)>, affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%98 : tensor<32x8x2x64xf32>) outs(%101 : tensor<32x2x64x8xf32>) {
    ^bb0(%in: f32, %out: f32):
      linalg.yield %in : f32
    } -> tensor<32x2x64x8xf32>
```

On my local generator, I also get:
```
error: 'tpp.identity' op fails to verify broadcasting rules
    %11 = tensor.empty() : tensor<10x13xf32>
    %12 = linalg.generic {indexing_maps = [#map4, #map3], iterator_types = ["parallel", "parallel"]} ins(%10 : tensor<10xf32>) outs(%11 : tensor<10x13xf32>) {
    ^bb0(%in: f32, %out: f32):
      linalg.yield %in : f32
    } -> tensor<10x13xf32>
```

So, this is not just the transpose identified in MHA, it's more widespread and needs attention.

@chelini

**Comment by chelini** [comment]
@nhasabni probably the issue is now fixed in https://github.com/plaidml/tpp-mlir/commit/0d8e72553300cff3ccf2e7f2a35294811f60c386 can you please confirm? If so feel free to close this issue.

**Comment by nhasabni** [comment]
Yes, I no longer see the broadcast-related failure in MHA.

**Comment by nhasabni** [comment]
> @nhasabni probably the issue is now fixed in [0d8e725](https://github.com/plaidml/tpp-mlir/commit/0d8e72553300cff3ccf2e7f2a35294811f60c386) can you please confirm? If so feel free to close this issue.

Yes, I no longer see the broadcast-related failure in MHA.
